src/matter/mdns/MdnsBroadcaster.ts

```diff
@@ -110,7 +112,9 @@ export class MdnsBroadcaster implements Broadcaster {
                 if (isIPv4(ip)) {
                     records.push(ARecord(hostname, ip));
                 } else {
-                    records.push(AAAARecord(hostname, ip));
+                    if(ip.startsWith('fe80::')){ // only add the local address
```

Comment:
Is this the right place? What about Interfaces with just non local IPv6? They would still add the other records, or?! Makes that sense?

Reply:
It makes sense, but my reading of the spec - which might be incorrect - is that only local IPv6 should be used for the IPv6 Unicast & Multicast address per section 2.5.6.2 and 2.5.6.1 in the Core Spec.

Reply:
I agree with the change. it is just the wrong code place in my eyes. We shoukd sort it out directly when getting the networtk interfaces and maybe then cut out whole interfaces if we removed all adresses

Reply:
There could be twoplaces where this better matches.
1.) In https://github.com/mfucci/node-matter/blob/main/src/net/node/NetworkNode.ts#L72 we prepare mac and ip list for a network interface ... that is currently also only used when broadcasting when I digged through code correctly
2.) In https://github.com/mfucci/node-matter/blob/main/src/net/MdnsServer.ts#L80 we prepare the list of network interfaces for announce. Here we shuld only return those with an IP in it that is announceable

So maybe idea is to sort out all "non local ipv6" in place #1 and check that this has no effects for receiving, but should not. Then sort out networkinterfaces without any IP left in  #2and throw error when nothing is left?!

Reply:
Thanks for the pointers - I'll do some testing with it in MdnsServer as that is closer than where I originally put the logic.  cheers.

Reply:
Looks like NetworkNode makes the most sense.  Note that I have only tested against HomeApp, but I validated that the IPAddrs for two TP-Link production Matter devices only have the `fe`, `fd` prefixed IPv6 addrs in the _matter._tcp entries AND in re-reading the 2.5.6.2 section of the Core Spec believe this is correct.  Please sanity check this.

Reply:
"fe" is link.local, fc/fc is site local ...(and yes fc deprecated from what I read, but could still be out there)

Reply:
ooookkkk ... I had a longer chat with a Network administrator about the whole topic and ... well ... it's not that easy :-)

IF we assume that all users have only one "network segment"/"VLAN" then fe would be best because we would know that all devices are "Local" and so in the same network. BUT I think we can not assume this safely because alone I know enough people that separate "iot stuff" into an own network and all other stuff into s separate one. Yes they will then have fun that UDP flows correctly, but thats another story.

But annoucement wise it makes no sense, from what I understood, to limit network interfaces at all because it is still valid to use global adresses for better routing ...

Yes maybe we need "site local only" for group multicsts later ... but thats later".

Do anyone know opther people to get expertises? I will also try to find the codeplace in chip tool later ... lets see what they do